File: workspace/workspace-fee-crawler/scripts/check/check_sheets.py
"""
스프레드시트의 모든 시트 이름 확인
"""
from google.oauth2 import service_account
from googleapiclient.discovery import build

# 설정
SPREADSHEET_ID = '1njdeOI4TLyF2IkggosBUGgg5yKetez8cdcepbsAeEx4'
SERVICE_ACCOUNT_FILE = './src/config/google_api_key.json'
SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']

# 인증
credentials = service_account.Credentials.from_service_account_file(
    SERVICE_ACCOUNT_FILE, scopes=SCOPES)
service = build('sheets', 'v4', credentials=credentials)

# 스프레드시트 메타데이터 가져오기
spreadsheet = service.spreadsheets().get(spreadsheetId=SPREADSHEET_ID).execute()

print("=" * 80)
print("스프레드시트의 모든 시트 목록")
print("=" * 80)

sheets = spreadsheet.get('sheets', [])
for i, sheet in enumerate(sheets, 1):
    properties = sheet.get('properties', {})
    sheet_id = properties.get('sheetId')
    sheet_name = properties.get('title')
    print(f"{i}. {sheet_name} (ID: {sheet_id})")

print("\n총 시트 수:", len(sheets))

File: workspace/confluence/scripts/check-sheets.js
#!/usr/bin/env node
import { google } from 'googleapis';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import { dirname } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const SPREADSHEET_ID = '14x_3ry6R9gNwvrmqLvnVEb8219fZ76B5h98eZrPcR64';
const CREDENTIALS_PATH = path.resolve(__dirname, '../../../config/google_api_key.json');

async function main() {
  const credentials = JSON.parse(fs.readFileSync(CREDENTIALS_PATH, 'utf8'));
  const auth = new google.auth.GoogleAuth({
    credentials,
    scopes: ['https://www.googleapis.com/auth/spreadsheets']
  });

  const sheets = google.sheets({ version: 'v4', auth });

  const response = await sheets.spreadsheets.get({
    spreadsheetId: SPREADSHEET_ID
  });

  console.log('=== 스프레드시트 정보 ===\n');
  console.log('제목:', response.data.properties.title);
  console.log();

  console.log('=== 시트 목록 ===\n');
  response.data.sheets.forEach(sheet => {
    console.log(`이름: ${sheet.properties.title}`);
    console.log(`ID: ${sheet.properties.sheetId}`);
    console.log(`색인: ${sheet.properties.index}`);
    console.log();
  });
}

main();
